node/peer.js

```diff
@@ -205,30 +205,35 @@ function waitForIdentified(callback) {
 
     if (conn.closing) {
         callback(conn.closeError);
-    } else if (conn.remoteName) {
+        return;
+    }
+
+    if (conn.remoteName) {
         callback(null);
-    } else {
-        self._waitForIdentified(conn, callback);
```

Comment:
There was a reason for this.

I do not want to allocate closures in `waitForIdentified()` and moved the closure allocation into a seperate function.